Formula/stormchecker.rb

```diff
@@ -1,28 +1,25 @@
 class Stormchecker < Formula
   desc "A modern probabilistic model checker"
   homepage "https://www.stormchecker.org"
-  url "https://github.com/moves-rwth/storm/archive/1.7.0.tar.gz"
-  version "1.7.0"
-  sha256 "e3ab398a6497059058408b6d09108e7c224da7fe1515c9e57f63b7e1fd95399c"
-  head "https://github.com/moves-rwth/storm.git", :using => :git
+  url "https://github.com/moves-rwth/storm/archive/1.8.0.tar.gz"
+  version "1.8.0"
+  sha256 "ea4467dfc3e984b7b707d2d8f73988049e9ca5f4489738fa067caa0ba0796e35"
+  head "https://github.com/moves-rwth/storm.git", using: :git, branch: "master"
 
   # option "with-single-thread", "Build Storm using just one thread."
   option "with-tbb", "Build Storm with Intel Thread Building Blocks (TBB) support."
-  option "with-cocoalib", "Build with support for CoCoALib (also requires CArl to be built with support for CoCoALib)."
   option "with-spot", "Build Storm with Spot (required for LTL model checking)."
 
   depends_on "autoconf" => :build
   depends_on "automake" => :build
   depends_on "cmake" => :build
   depends_on "boost"
-  depends_on "doxygen"
   depends_on "glpk"
   depends_on "gmp"
   depends_on "hwloc"
-  depends_on "moves-rwth/misc/carl" => build.with?("cocoalib") ? ["with-thread-safe", "with-cln", "with-ginac", "with-cocoalib"] : ["with-thread-safe", "with-cln", "with-ginac"]
+  depends_on "moves-rwth/misc/carl-storm" => ["with-cln", "with-ginac"]
```

Comment:
Do we need something like this?
```suggestion
  on_intel do
    depends_on "moves-rwth/misc/carl-storm" => ["with-cln", "with-ginac"]
  end
  on_arm do
    depends_on "moves-rwth/misc/carl-storm"
  end
```
(Disclaimer: I haven't tested this but I can test it on my arm machine if that helps)

Reply:
I think the `carl-storm` formula already takes care of the correct settings. But it might make sense to explicitly specify the optional parameters in this formula as well.

I only tested the formula on my machine with an Intel CPU. It would be great if you or somebody else could try it on ARM (M1) as well.

Reply:
I tried the formula (without my suggested change) on my ARM machine and things worked perfectly.
I approve merging this.

For the record: It does consider the carl-storm formula with `--with-cln --with-ginac` flags but I guess those are simply ignored.
```console
!master ~/git/homebrew-storm/Formula> brew info carl-storm
==> moves-rwth/misc/carl-storm: stable 14.25, HEAD
Computer ARithmetic and Logic library for the probabilistic model checker Storm
https://ths-rwth.github.io/carl/
/opt/homebrew/Cellar/carl-storm/14.25 (287 files, 2.8MB) *
  Built from source on 2023-06-02 at 12:05:23
From: https://github.com/moves-rwth/homebrew-misc/blob/HEAD/Formula/carl-storm.rb
==> Dependencies
Required: boost ✔, cmake ✔, eigen ✔, gmp ✔
==> Options
--HEAD
	Install HEAD version
!master ~/git/homebrew-storm/Formula> brew info stormchecker           
==> stormchecker: stable 1.8.0, HEAD
A modern probabilistic model checker
https://www.stormchecker.org
/opt/homebrew/Cellar/stormchecker/1.8.0 (1,589 files, 77.9MB) *
  Built from source on 2023-06-02 at 12:10:15
From: /opt/homebrew/opt/stormchecker/.brew/stormchecker.rb
==> Dependencies
Build: autoconf ✔, automake ✔, cmake ✔
Required: boost ✔, glpk ✔, gmp ✔, hwloc ✔, moves-rwth/misc/carl-storm --with-cln --with-ginac ✔, xerces-c ✔, z3 ✔
Optional: spot ✔, tbb ✘
==> Options
--with-spot
	Build Storm with Spot (required for LTL model checking).
--with-tbb
	Build Storm with Intel Thread Building Blocks (TBB) support.
--HEAD
	Install HEAD version
```

Reply:
Thanks for testing it. I followed your suggestion after all and now distinguish between Intel and ARM. Otherwise it becomes too confusing.
In your case `carl-storm` is listed with both options `--with-cln --with-ginac` even though these options were ignored. I think that was too misleading.

Reply:
Thanks! Still works for me

As far as I can see, carl is build with the right options
```console
cmake /private/tmp/carl-storm-20230605-51329-js9bs9/carl-storm-14.25 -DCMAKE_INSTALL_PREFIX=/opt/homebrew/Cellar/carl-storm/14.25 -DCMAKE_INSTALL_LIBDIR=lib -DCMAKE_BUILD_TYPE=Release -DCMAKE_FIND_FRAMEWORK=LAST -DCMAKE_VERBOSE_MAKEFILE=ON -Wno-dev -DBUILD_TESTING=OFF -DCMAKE_OSX_SYSROOT=/Library/Developer/CommandLineTools/SDKs/MacOSX13.sdk -DEXPORT_TO_CMAKE=OFF -DCMAKE_BUILD_TYPE=RELEASE -DEXCLUDE_TESTS_FROM_ALL=ON -DTHREAD_SAFE=ON -DUSE_CLN_NUMBERS=OFF -DUSE_GINAC=OFF
```

And stormchecker no longer lists carl-storm with cln and ginac flags:
```console
brew info stormchecker
==> stormchecker: stable 1.8.0, HEAD
A modern probabilistic model checker
https://www.stormchecker.org
/opt/homebrew/Cellar/stormchecker/1.8.0 (1,589 files, 77.9MB) *
  Built from source on 2023-06-05 at 13:49:39
From: /opt/homebrew/opt/stormchecker/.brew/stormchecker.rb
==> Dependencies
Build: autoconf, automake, cmake
Required: boost, glpk, gmp, hwloc, moves-rwth/misc/carl-storm, xerces-c, z3
Optional: spot, tbb
==> Options
--with-spot
	Build Storm with Spot (required for LTL model checking).
--with-tbb
	Build Storm with Intel Thread Building Blocks (TBB) support.
--HEAD
	Install HEAD version
```

Reply:
Looks good. Thanks for your help!